### ISSUE TITLE
FixedDenseConnections.indices not saved in state_dict()

### ISSUE THREAD

**Issue by Arash-ethz-1** [ligerlac/torchlogix#22]
## Root Cause

`FixedDenseConnections` stores its input wiring tensor as a plain Python attribute:

```python
# connections.py, line 105
self.indices = self._init_connections()
```

PyTorch's `state_dict()` only captures `nn.Parameter` objects and registered buffers. Plain attributes are silently ignored, meaning `indices` is **never included** when saving with:

```python
torch.save(model.state_dict(), path)
```

On reload, a new model instance is created and new random connections are generated. `load_state_dict()` restores the weights but leaves the connections randomized. The loaded model now has weights optimized for one wiring applied to a completely different wiring, producing essentially random outputs.

---

## Symptoms

- Model reports correct accuracy during training (connections live in memory throughout).
- After save + reload, accuracy collapses to near-random (in our case: **88% → ~50%**, win rate **0%** in Minesweeper).
- Bug is **silent**.

---

## Reproduction

```python
model = LogicDense(in_dim=300, out_dim=2048, lut_rank=4, connections="fixed")
x = torch.randn(8, 300)
out_before = model(x)

torch.save(model.state_dict(), "model.pt")

model2 = LogicDense(in_dim=300, out_dim=2048, lut_rank=4, connections="fixed")
model2.load_state_dict(torch.load("model.pt"))

# Different output — wrong connections after reload
assert torch.allclose(out_before, model2(x))  # FAILS
```

---

## Fix

```python
# connections.py
self.register_buffer('indices', self._init_connections())
```

---

## Why It Wasn't Caught

The bug is not visible in `experiments/train.py` because evaluation is always done **in-process** without reloading. The only reload in the codebase (`generate_test_vectors.py`) correctly uses `torch.load(model)` (full object pickle), which **does** preserve plain attributes, masking the issue entirely.

**Comment by ligerlac** [comment]
Dear @Arash-ethz-1 , thanks for spotting this bug. The fix really seems to be this one-liner. Feel free to make a PR, otherwise I'll fix it later this week.
